scripts/experiments_images/ocsvm.jl

```diff
@@ -0,0 +1,112 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using GenerativeAD.Models
+using BSON
+using StatsBase: fit!, predict, sample
+
+using MLDataPattern
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+		required = true
+		arg_type = Int
+		help = "seed"
+	"dataset"
+		required = true
+		arg_type = String
+		help = "dataset"
+	"anomaly_classes"
+		arg_type = Int
+		default = 10
+		help = "number of anomaly classes"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed, anomaly_classes = parsed_args
+
+modelname = "OCSVM"
+
+function sample_params()
+	par_vec = (round.([10^x for x in -4:0.1:2],digits=5),["poly", "rbf", "sigmoid"])
+	argnames = (:gamma,:kernel)
+	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+end
+
+function joint_fit(models, data_splits)
+    info = []
+    for (model, data) in zip(models, data_splits)
+        push!(info, fit!(model, data))
+    end
+    return info
+end
+
+function joint_prediction(models, data)
+    joint_pred = Array{Float32}(undef, length(models), size(data,2))
+    for (i,model) in enumerate(models)
```

Comment:
Is it not more economical to do the prediction here sample-wise? That way you only need to allocate a vector as you can already do the average in each step.